eclipse/che-devfile-registry
# chore(devfile): Add first dog fooding devfile

### What does this PR do?
A beginning of a dogfooding devfile to work on the devfile registry:

Provides containers and che commands to
- build in the workspace
- test the registry in a dedicated container in the workspace
- publish to surge for live update testing 

Created https://github.com/eclipse/che/issues/19305 to continue:
- investigate 'start devfile-registry' task is failing when running it from a che-theia task.
- build the offline version
- didact guide
- badge in README.md

### Screenshot/screencast of this PR

![Selection_272](https://user-images.githubusercontent.com/650571/111446633-e702a680-870c-11eb-9ed5-0841ea45cfca.png)
![Selection_274](https://user-images.githubusercontent.com/650571/111446610-e10cc580-870c-11eb-89a7-36e63df825c9.png)


### What issues does this PR fix or reference?
N/A


### How to test this PR?
You can try the live change flow:

1. Use the factory https://workspaces.openshift.com/f?url=https://github.com/eclipse/che-devfile-registry/tree/devfile
2. make a change in `/projects/che-devfile-registry/devfiles` folder
3. run the build che-command
4. run the publish to surge.sh command
5. give a try in your own che-server changing your CheCluster CR:
    ```yaml
       devfileRegistryUrl: https://sunix-che-devfile-registry-lombok-qpd6g.surge.sh
       externalDevfileRegistry: true
    ```
7. refresh the dashboard `Get Started Page` and see your changes
5. make a change in `/projects/che-devfile-registry/devfiles` folder
6. run the build che-command
7. run the publish to surge.sh
8. refresh the dashboard `Get Started Page` and see the live changes

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)


### Reviewers

Reviewers, please comment how you tested the PR when approving it.
